lib/solr.ts

```diff
@@ -64,6 +66,7 @@ export class Client {
     this.REAL_TIME_GET_HANDLER = 'get';
     this.SPELL_HANDLER = 'spell';
     this.TERMS_HANDLER = 'terms';
+    this.logger = options.logger ?? console;
```

Comment:
I think this should default to NO logging, and we should type it in such a way that passing in `console` would satisfy the type, but also make it simple enough to roll your own. I'm thinking of plugging a little function that calls Sentry here for one of my projects...

Reply:
sure, I'll replace it with noop

Reply:
Hmm, also, now we are storing some stuff inside `this.options` and some of it outside. And yet the `options` argument and `this.options` have the same type.

Reply:
Types are different. Also I believe the way options are currently stored is wrong and messy, but that would be a larger refactoring. Also I have no idea why we store constants as class fields. Logger doesn't sound like it should be stored as an `options` thing.

Reply:
Hmm, I see. You are right. It's quite messy.